modopt/tests/test_signal.py

```diff
@@ -1,322 +1,233 @@
-
-        npt.assert_raises(TypeError, positivity.positive, '-1')
+    @pytest.fixture
+    def svd0(self, data):
+        """Compute SVD of first data sample."""
+        return svd.calculate_svd(data[0])
 
-
-class SVDTestCase(TestCase):
-    """Test case for svd module."""
-
-    def setUp(self):
-        """Set test parameter values."""
-        self.data1 = np.arange(18).reshape(9, 2).astype(float)
-        self.data2 = np.arange(32).reshape(16, 2).astype(float)
-        self.data3 = np.array(
-            [
-                np.array([
-                    [-0.01744594, -0.61438865],
-                    [-0.08435304, -0.50397984],
-                    [-0.15126014, -0.39357102],
-                    [-0.21816724, -0.28316221],
-                    [-0.28507434, -0.17275339],
-                    [-0.35198144, -0.06234457],
-                    [-0.41888854, 0.04806424],
-                    [-0.48579564, 0.15847306],
-                    [-0.55270274, 0.26888188],
-                ]),
-                np.array([42.23492742, 1.10041151]),
-                np.array([
-                    [-0.67608034, -0.73682791],
-                    [0.73682791, -0.67608034],
-                ]),
-            ],
-            dtype=object,
-        )
-        self.data4 = np.array([
-            [-1.05426832e-16, 1.0],
-            [2.0, 3.0],
-            [4.0, 5.0],
-            [6.0, 7.0],
-            [8.0, 9.0],
-            [1.0e1, 1.1e1],
-            [1.2e1, 1.3e1],
-            [1.4e1, 1.5e1],
-            [1.6e1, 1.7e1],
-        ])
-        self.data5 = np.array([
-            [0.49815487, 0.54291537],
-            [2.40863386, 2.62505584],
-            [4.31911286, 4.70719631],
-            [6.22959185, 6.78933678],
-            [8.14007085, 8.87147725],
-            [10.05054985, 10.95361772],
-            [11.96102884, 13.03575819],
-            [13.87150784, 15.11789866],
-            [15.78198684, 17.20003913],
-        ])
-        self.svd = svd.calculate_svd(self.data1)
-
-    def tearDown(self):
-        """Unset test parameter values."""
-        self.data1 = None
-        self.data2 = None
-        self.data3 = None
-        self.data4 = None
-        self.svd = None
-
-    def test_find_n_pc(self):
-        """Test find_n_pc."""
+    def test_find_n_pc(self, data):
+        """Test find number of principal component."""
         npt.assert_equal(
-            svd.find_n_pc(svd.svd(self.data2)[0]),
+            svd.find_n_pc(svd.svd(data[1])[0]),
             2,
-            err_msg='Incorrect number of principal components.',
+            err_msg="Incorrect number of principal components.",
         )
 
+    def test_n_pc_fail_non_square(self):
+        """Test find_n_pc."""
         npt.assert_raises(ValueError, svd.find_n_pc, np.arange(3))
 
-    def test_calculate_svd(self):
+    def test_calculate_svd(self, data, svd0):
         """Test calculate_svd."""
+        errors = []
+        for i, name in enumerate("USV"):
```

Comment:
This is really not a big deal and just a minor style choice, but I am pushing for ban on the use of the variable `i` as a loop index. In general I would like to avoid single character variables in general. Anyway, I would propose `index` or `idx` instead. Feel free to leave it, just keep in mind that I will 100% change it later. 😉

Reply:
I dare you :stuck_out_tongue: